integration_tests/src/main/python/join_test.py

```diff
@@ -397,17 +397,44 @@ def do_join(spark):
         return left.join(broadcast(right), left.a > f.log(right.r_a), join_type)
     assert_gpu_and_cpu_are_equal_collect(do_join)
 
+@allow_non_gpu('BroadcastExchangeExec')
+@ignore_order(local=True)
+@pytest.mark.parametrize('data_gen', [IntegerGen(), LongGen(), pytest.param(FloatGen(), marks=[incompat]), pytest.param(DoubleGen(), marks=[incompat])], ids=idfn)
+@pytest.mark.parametrize('join_type', ['Cross', 'Right'], ids=idfn)
+def test_broadcast_nested_loop_join_with_non_ast_condition_no_fallback(data_gen, join_type):
+    def do_join(spark):
+        left, right = create_df(spark, data_gen, 50, 25)
+        # AST does not support cast or logarithm yet which is supposed to be extracted into child
+        # nodes
+        return broadcast(left).join(right, f.round(left.a).cast('integer') > f.round(f.log(right.r_a).cast('integer')), join_type)
```

Comment:
Yes, it's intentionally changed since AST didn't support comparison. https://github.com/NVIDIA/spark-rapids/blob/25ffa873ad6211f14d5e402805e537b52d60d2dc/sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala#L756

The previous NaN problem is fixed in my latest [update](https://github.com/NVIDIA/spark-rapids/pull/9635/commits/6546c3cf6e9d1be06d6870e999db58261309e0aa#diff-6d842883ea07751fad4e05bd48a8aa750f25319b74a65b1a200938b4c2866addR41-R42). Actually it's a bug that passed the check ```(double_a_from_tbl1 > double_b_from_tbl2)``` which should not be allowed since it's crossing multiple sides. The root cause here is we should compare with exprID other than attribute. In that particular case, there is no NotNull child after scan node. So the attribute is changed to nullable=false while original expression is still nullable=true. It will be different when we do the comparison.